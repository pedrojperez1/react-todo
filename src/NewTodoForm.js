import React, {useState} from "react";
import {v4 as uuid} from "uuid";

const NewTodoForm = ({add}) => {
    const INITIAL_STATE = {
        todo: ''
    };
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        e.preventDefault();
        const {name, value} = e.target;
        setFormData(formData => (
            {...formData, [name]: value}
        ))
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        const newTodo = {
            ...formData, id: uuid()
        }
        add(newTodo);
        setFormData(INITIAL_STATE);
    }
    return (
        <div>
            <form onSubmit={handleSubmit}>
                <input 
                    type="text" 
                    placeholder="New Todo" 
                    name="todo"
                    onChange={handleChange}
                    value={formData.todo}
                ></input>
                <button>Add</button>
            </form>
        </div>
    )
};

export default NewTodoForm;