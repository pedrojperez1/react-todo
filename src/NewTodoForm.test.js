import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("should render without going up in smoke", () => {
    render(<NewTodoForm />);
});
  
it("should render the same every time", () => {
  const {asFragment} = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
  