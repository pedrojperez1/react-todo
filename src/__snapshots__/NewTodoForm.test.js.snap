// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same every time 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="todo"
        placeholder="New Todo"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
